cli/src/main.rs

```diff
@@ -0,0 +1,375 @@
+use std::path::PathBuf;
+
+use anchor_client::solana_sdk::bpf_loader_upgradeable;
+use anchor_client::solana_sdk::commitment_config::CommitmentConfig;
+use anchor_client::solana_sdk::instruction::Instruction;
+use anchor_client::solana_sdk::pubkey::Pubkey;
+use anchor_client::solana_sdk::signature::read_keypair_file;
+use anchor_client::solana_sdk::signature::{Keypair, Signer};
+use anchor_client::solana_sdk::system_instruction;
+use anchor_client::solana_sdk::sysvar;
+use anchor_client::{Client, Cluster, Program};
+use clap::Clap;
+use multisig::accounts as multisig_accounts;
+use multisig::instruction as multisig_instruction;
+use rand::rngs::OsRng;
+
+/// Multisig -- interact with a deployed Multisig program.
+#[derive(Clap, Debug)]
+struct Opts {
+    /// Address of the Multisig program.
+    #[clap(long)]
+    multisig_program_id: Pubkey,
+
+    /// The keypair to sign and pay with. Defaults to ~/.config/solana/id.json.
+    #[clap(long)]
+    keypair_path: Option<PathBuf>,
+
+    #[clap(subcommand)]
+    subcommand: SubCommand
+}
+
+#[derive(Clap, Debug)]
+enum SubCommand {
+    /// Create a new multisig address.
+    CreateMultisig(CreateMultisigOpts),
```

Comment:
It’s nice to have the structs anyway because we can pass them as arguments to the handlers for every command, without having to pattern match and unpack and then repack them into function call arguments.